singularity-eos/eos/eos_base.hpp

```diff
@@ -66,23 +66,24 @@ char *StrCat(char *destination, const char *source) {
 // VECTOR functionality to overload the scalar implementations in the derived
 // classes. Do not add functions here that are not overloads of derived class features.
 // TODO(JMM): Should we have more macros that capture just some of these?
-#define SG_ADD_BASE_CLASS_USINGS(EOSDERIVED)                                             \
-  using EosBase<EOSDERIVED>::TemperatureFromDensityInternalEnergy;                       \
-  using EosBase<EOSDERIVED>::InternalEnergyFromDensityTemperature;                       \
-  using EosBase<EOSDERIVED>::PressureFromDensityTemperature;                             \
-  using EosBase<EOSDERIVED>::PressureFromDensityInternalEnergy;                          \
-  using EosBase<EOSDERIVED>::MinInternalEnergyFromDensity;                               \
-  using EosBase<EOSDERIVED>::SpecificHeatFromDensityTemperature;                         \
-  using EosBase<EOSDERIVED>::SpecificHeatFromDensityInternalEnergy;                      \
-  using EosBase<EOSDERIVED>::BulkModulusFromDensityTemperature;                          \
-  using EosBase<EOSDERIVED>::BulkModulusFromDensityInternalEnergy;                       \
-  using EosBase<EOSDERIVED>::GruneisenParamFromDensityTemperature;                       \
-  using EosBase<EOSDERIVED>::GruneisenParamFromDensityInternalEnergy;                    \
-  using EosBase<EOSDERIVED>::FillEos;                                                    \
-  using EosBase<EOSDERIVED>::EntropyFromDensityTemperature;                              \
-  using EosBase<EOSDERIVED>::EntropyFromDensityInternalEnergy;                           \
-  using EosBase<EOSDERIVED>::GibbsFreeEnergyFromDensityTemperature;                      \
-  using EosBase<EOSDERIVED>::GibbsFreeEnergyFromDensityInternalEnergy;
+// JMM: Use VA_ARGS to capture more complex template types
+#define SG_ADD_BASE_CLASS_USINGS(...)                                                    \
```

Comment:
The problem here is that preprocessor macros interpret commas as separate arguments. This allows us to pass classes with multiple template arguments through the macro transparently.